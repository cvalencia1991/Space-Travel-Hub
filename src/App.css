.stylesjoin {
  width: 10%;
}

.active-link {
  text-decoration: underline;
}

.link {
  text-decoration: none;
}
