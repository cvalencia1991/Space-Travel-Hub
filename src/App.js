/* eslint-disable react-hooks/exhaustive-deps */
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import NavbarSpace from './Components/Navbar';
import './App.css';
import Myprofile from './Components/Myprofile';
import Rockets from './Components/Rockets';
import Missions from './Components/Missions';
import { getRockets } from './Redux/Rockets/rocketsSlice';
import { getMissions } from './Redux/Missions/Missionslice';

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRockets(), getMissions());
  }, []);

  return (
    <Router>
      <NavbarSpace />
      <Routes>
        <Route exact path="/" element={<Myprofile />} />
        <Route path="/rockets" element={<Rockets />} />
        <Route exact path="/Missions" element={<Missions />} />
      </Routes>
    </Router>
  );
}

export default App;
