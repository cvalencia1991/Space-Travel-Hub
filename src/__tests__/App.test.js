import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../Redux/configureStore';

it('renders correctly', () => {
  const app = renderer
    .create(<Provider store={store}><App /></Provider>)
    .toJSON();
  expect(app).toMatchSnapshot();
});
