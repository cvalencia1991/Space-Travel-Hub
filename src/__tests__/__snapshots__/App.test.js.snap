// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="p-4 navbar navbar-expand navbar-light"
  >
    <div
      className="container"
    >
      <a
        className="d-flex align-items-center navbar-brand"
        href="/"
      >
        <img
          alt="Space Travel Hublogo"
          className="d-inline-block align-top"
          height="80"
          src="planet.png"
          width="80"
        />
        <h1
          className="d-inline ms-2"
        >
          Space Traveler's Hub
        </h1>
      </a>
      <ul
        className="d-flex p-0 gap-3 navigation"
        style={
          Object {
            "listStyle": "none",
          }
        }
      >
        <li>
          <a
            aria-current="page"
            className="active-link"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className="link"
            href="/Missions"
            onClick={[Function]}
          >
            Missions
          </a>
        </li>
        |
        <li>
          <a
            className="link"
            href="/myProfile"
            onClick={[Function]}
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="rocket-list m-5 p-3"
  />,
]
`;
